Holovibes/sources/gui/windows/panels/export_panel.cc

```diff
@@ -45,34 +44,77 @@ void actualise_record_output_file_ui(const std::filesystem::path file_path)
+            ui_->RecordExtComboBox->setCurrentIndex(extension_indexes[0]);
     }
 
-    if (ui_->TimeTransformationCutsCheckBox->isChecked())
+    if (api::get_record_mode() == RecordMode::CHART)
     {
-        // Only one check is needed
-        if (ui_->RecordImageModeComboBox->findText("3D Cuts XZ") == -1)
+        ui_->ChartPlotWidget->show();
+
+        if (gui::get_main_display())
         {
-            ui_->RecordImageModeComboBox->insertItem(1, "3D Cuts XZ");
-            ui_->RecordImageModeComboBox->insertItem(1, "3D Cuts YZ");
+            gui::get_main_display()->resetTransform();
+
+            gui::get_main_display()->getOverlayManager().enable<Noise>();
+            gui::get_main_display()->getOverlayManager().enable<Signal>();
         }
     }
     else
     {
-        ui_->RecordImageModeComboBox->removeItem(ui_->RecordImageModeComboBox->findText("3D Cuts XZ"));
-        ui_->RecordImageModeComboBox->removeItem(ui_->RecordImageModeComboBox->findText("3D Cuts YZ"));
+        ui_->ChartPlotWidget->hide();
+
+        if (gui::get_main_display())
+        {
+            gui::get_main_display()->resetTransform();
+
+            gui::get_main_display()->getOverlayManager().disable(Signal);
+            gui::get_main_display()->getOverlayManager().disable(Noise);
+        }
     }
 
+    // Record type
+    auto img_mode_view = qobject_cast<QListView*>(ui_->RecordImageModeComboBox->view());
+
+    static std::map<RecordMode, QString> record_mode_map = {{RecordMode::RAW, "Raw Image"},
+                                                            {RecordMode::HOLOGRAM, "Processed Image"},
+                                                            {RecordMode::MOMENTS, "Moments"},
+                                                            {RecordMode::CHART, "Chart"},
+                                                            {RecordMode::CUTS_XZ, "3D Cuts XZ"},
+                                                            {RecordMode::CUTS_YZ, "3D Cuts YZ"}};
+    ui_->RecordImageModeComboBox->setCurrentIndex(
+        ui_->RecordImageModeComboBox->findText(record_mode_map[api::get_record_mode()]));
+
+    // Hiding most of the options when in raw mode
+    const bool is_raw = api::get_compute_mode() == Computation::Raw;
+    img_mode_view->setRowHidden(ui_->RecordImageModeComboBox->findText("Processed Image"), is_raw);
+    img_mode_view->setRowHidden(ui_->RecordImageModeComboBox->findText("Moments"), is_raw);
+    img_mode_view->setRowHidden(ui_->RecordImageModeComboBox->findText("Chart"), is_raw);
+
+    // When set to raw, the 3D cuts are automatically disabled, so this works as a valid toggle for raw mode too
+    const bool hide_cuts = !ui_->TimeTransformationCutsCheckBox->isChecked();
+    img_mode_view->setRowHidden(ui_->RecordImageModeComboBox->findText("3D Cuts XZ"), hide_cuts);
+    img_mode_view->setRowHidden(ui_->RecordImageModeComboBox->findText("3D Cuts YZ"), hide_cuts);
```

Comment:
Same here instead of find text you can just static cast the right enum field (if reordered)